agent/connect/ca/provider_vault.go

```diff
@@ -529,12 +529,7 @@ func (v *VaultProvider) Sign(csr *x509.CertificateRequest) (string, error) {
 	if !ok {
 		return "", fmt.Errorf("certificate was not a string")
 	}
-	ca, ok := response.Data["issuing_ca"].(string)
-	if !ok {
-		return "", fmt.Errorf("issuing_ca was not a string")
-	}
-
-	return EnsureTrailingNewline(cert) + EnsureTrailingNewline(ca), nil
+	return EnsureTrailingNewline(cert), nil
```

Comment:
There is more context in the commit message, but I believe this change is correct because it aligns the vault provider with the other providers, and because we append the full chain to this cert in the Sign RPC endpoint. So previous it would have been adding the same CA cert twice. 

That was fine before when it was just a single root cert, but it produces an unexpected chain when we have multiple certs in the chain. The tests added in subsequent commits should exercise this code extensively in both primary and secondary DC.